pymine/logic/parsers/minecraft/vec2.py

```diff
@@ -0,0 +1,20 @@
+from pymine.api.errors import ParsingError
+from pymine.api.abc import AbstractParser
+from pymine.types.player import Player
+
+class Vec2(AbstractParser):
+    def __init__(self) -> None:
+        pass
+
+    def parse(self, s: str, p:[Player, None]) -> tuple:
```

Comment:
If a type can be None, you should not typehint it like that. It should be more like `foo: bar =  None`, in addition the player object should probably be required, so no need to make it an optional arg.